src/NHibernate/Type/SingleType.cs

```diff
@@ -62,7 +62,7 @@ public override System.Type ReturnedClass
 
 		public override void Set(DbCommand rs, object value, int index, ISessionImplementor session)
 		{
-			rs.Parameters[index].Value = value;
+			rs.Parameters[index].Value = Convert.ToSingle(value);
```

Comment:
Maybe it should be  `Convert.ToSingle(value, CultureInfo.InvariantCulture)`?

Reply:
Normally the argument is a numeric type, so I am not sure whether `CultureInfo` is even used in that case. If we add the invariant culture for `Single` then we should do it also for other types to be consistent, which I think should be done in a separate PR.

Reply:
Ok